### PR TITLE
Use Nox to drive MacOS tests

### PR DESCRIPTION
**Pull request by s0undt3ch**
### What does this PR do?
See title.

### CODE REVIEW

**Review by Akm0d — APPROVED** [review]

File: .ci/kitchen-macosxhighsierra-py2 [hunk 1]
```diff
@@ -14,6 +14,9 @@ def golden_images_branch = 'master'
 def nox_passthrough_opts = ''
 def concurrent_builds = 1
 def jenkins_slave_label = 'kitchen-slave-mac'
+def kitchen_driver_file = '/var/jenkins/workspace/driver.yml'
+def kitchen_verifier_file = '/var/jenkins/workspace/nox-verifier.yml'
+def kitchen_platforms_file = '/var/jenkins/workspace/pre-golden-nox-platforms.yml'
```
Comment (bryceml) [inline .ci/kitchen-macosxhighsierra-py2]: why not call it nox-platforms.yml so it's the same as the others?  just wondering.

Comment (s0undt3ch) [reply .ci/kitchen-macosxhighsierra-py2]: We already have that one setup in sre-jenkins, but for proper MacOS golden images.

Comment (s0undt3ch) [reply .ci/kitchen-macosxhighsierra-py2]: Just named it differently so that proper golden images can continue without affecting this switch to nox.

Comment (bryceml) [reply .ci/kitchen-macosxhighsierra-py2]: what is different between the files?

**Review by waynew — APPROVED** [review]
Aside from Bryce's valid nit, 👍

File: .ci/kitchen-macosxhighsierra-py2 (hunk 1 above)
Comment (s0undt3ch) [reply .ci/kitchen-macosxhighsierra-py2]: Proper golden images, [this](https://github.com/saltstack/sre-jenkins/blob/develop/states/services/jenkins-node/mac-install.sls#L457-L461), pre golden images [this](https://github.com/saltstack/sre-jenkins/blob/develop/states/services/jenkins-node/mac-install.sls#L540-L541) and [this](https://github.com/saltstack/sre-jenkins/blob/develop/states/services/jenkins-node/mac-install.sls#L566), as an example.

Comment (bryceml) [reply .ci/kitchen-macosxhighsierra-py2]: nice

**Review by bryceml — APPROVED** [review]

**Comment by s0undt3ch** [comment]
Follow up on https://github.com/saltstack/salt/pull/55272

**Pull request closed without merging**